eox_tagging/models.py

```diff
@@ -47,35 +50,49 @@ def invalid(self):
         """Returns all invalid tags."""
         return self.exclude(invalidated_at=None)
 
-    def find_by_owner(self, owner):
+    def find_by_owner(self, owner_type, owner_id):
         """Returns all valid tags owned by owner_id."""
-        model = owner.__class__
-        ctype = ContentType.objects.get_for_model(model)
-
-        return self.valid().filter(
-            owner_type=ctype,
-            owner_object_id=owner.id,
-        )
-
-    def find_all_tags_for(self, target_object):
-        """Returns all valid on an object."""
-        if get_model_name(target_object) in OPAQUE_KEY_PROXY_MODEL_TARGETS:
-            model = OpaqueKeyProxyModel
-            target_id = OpaqueKeyProxyModel.objects.get(opaque_key=target_object.course_id).id
-        else:
-            model = target_object.__class__
-            target_id = target_object.id
-        ctype = ContentType.objects.get_for_model(model)
 
-        return self.valid().filter(
-            target_type=ctype,
-            target_object_id=target_id,
-        )
+        try:
+            owner, ctype = self.__get_object_for_this_type(owner_type, owner_id)
+        except TypeError:
+            return self.none()
+
+        return self.filter(owner_type=ctype, owner_object_id=owner.id,)
+
+    def find_all_tags_for(self, target_type, target_id):
+        """Returns all valid tags on an object."""
+        target_type = PROXY_MODEL_NAME if target_type in OPAQUE_KEY_PROXY_MODEL_TARGETS else target_type
+
+        try:
+            target, ctype = self.__get_object_for_this_type(target_type, target_id)
+        except TypeError:
```

Comment:
does this happen because it doesn't return a tuple ??? maybe raise a custom exception is better or catch here the `ObjectDoesNotExist`

Reply:
yes!! When I return none it raises TypeError. Ok let me see what I can do